src/cursor/cur_std.c

```diff
@@ -1175,13 +1179,78 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
         WT_ERR(__wt_config_gets(session, cfg, "bound", &cval));
         if (cval.len == 0)
             WT_ERR_MSG(session, EINVAL, "setting bounds must require the bound configuration set");
+
+        /* The cursor must have a key set to place the lower or upper bound. */
+        WT_ERR(__cursor_checkkey(cursor));
+
+        if (WT_STRING_MATCH("upper", cval.str, cval.len)) {
+            /*
+             * If the lower bounds are set, make sure that the upper bound is greater than the
+             lower
+             * bound.
+             */
+            WT_ERR(__wt_cursor_get_raw_key(cursor, &key));
+            if (F_ISSET(cursor, WT_CURSTD_BOUND_LOWER)) {
+                WT_ERR(__wt_compare(
+                  session, CUR2BT(cursor)->collator, &key, &cursor->lower_bound, &exact));
+                if (exact < 0)
+                    WT_ERR_MSG(session, EINVAL, "The upper bound overlaps with the lower bound");
```

Comment:
I would make sure the error message is the same for both cases that way its easier to handle at the MongoDB layer. "The provided cursor bounds are overlapping."